analysis/02_new_model_input/03.2_computing_stats.md

```diff
@@ -0,0 +1,208 @@
+    typhoon_metadata["landfalldate"] + "-" + typhoon_metadata["landfall_time"]
+)
+
+typhoons = list(typhoon_metadata["typhoon"].values)
+#%% Processing the data into an excel sheet
+time_frame = 48  # in half hours
+time_frame_6 = 12  # in half hours
+mov_window = 12  # in half hours
+before_landfall_h = 72  # how many hours before landfall to include
+after_landfall_h = 72  # how many hours before landfall to include
+
+num_intervals = (
+    math.floor((2 * 144 - time_frame) / mov_window) + 1
+)  # 144 to include 72hr before and after
+num_intervals_6 = (
+    math.floor((2 * 144 - time_frame_6) / mov_window) + 1
+)  # 144 to include 72hr before and after
+```
+
+```python
+# looping over all typhoons
+for stats in ["mean", "max"]:
+    df_rainfall_final = pd.DataFrame(
+        columns=["typhoon", "id", "Centroid", "rainfall_Total"]
+    )
+    for typ in typhoons:
+        print(typ)
+        # Getting typhoon info
+        df_info = typhoon_metadata[typhoon_metadata["typhoon"] == typ]
+        landfall = df_info["landfall_date_time"].values[0]
+        landfall = datetime.datetime.strptime(landfall, "%d-%m-%Y-%H:%M:%S")
+        # End date is landfall date
+        # Start date is 72 hours before landfall date
+        # end_date = landfall
+        end_date = landfall + datetime.timedelta(
+            hours=after_landfall_h
+        )  # landfall
+        # start_date = end_date - datetime.timedelta(hours=before_landfall_h)
+        start_date = landfall - datetime.timedelta(hours=before_landfall_h)
+        # Loading the data
+        df_rainfall = pd.read_csv(
+            processed_output_dir + typ + "_gridstats_" + stats + ".csv"
+        )
+        # df_rainfall["time"] = (
+        #     df_rainfall["date"].astype(str) + "_" + df_rainfall["start"]
+        # )
+        # df_rainfall = pd.pivot(
+        #     df_rainfall,
+        #     index=["id", "Centroid"],
+        #     columns=["time"],
+        #     values=[stats],
+        # )
+        # df_rainfall.columns = df_rainfall.columns.droplevel(0)
+        # df_rainfall.reset_index(inplace=True)
+        # Convert column names to date format
+        for col in df_rainfall.columns[2:]:
+            date_format = datetime.datetime.strptime(col, "%Y%m%d_%H:%M:%S")
+            df_rainfall = df_rainfall.rename(columns={col: date_format})
+
+        df_mean_rainfall = pd.DataFrame(
+            {"id": df_rainfall["id"], "Centroid": df_rainfall["Centroid"]}
+        )
+        available_dates_t = [
+            date
+            for date in df_rainfall.columns[2:]
+            if (date >= start_date) & (date < end_date)
+        ]
+        #########################################
+        df_mean_rainfall_6 = pd.DataFrame(
+            {
+                "id": df_rainfall["id"].astype(str),
+                "Centroid": df_rainfall["Centroid"],
+            }
+        )
+
+        for i in range(num_intervals_6):
+            start = start_date + datetime.timedelta(
```

Comment:
I feel like this whole part could be simplified and clarified by using e.g. [`pandas.rolling`](https://pandas.pydata.org/docs/reference/api/pandas.DataFrame.rolling.html)